implement ChiselUtestTester for utest.

sequencer opened the pull request.

edwardcwang commented on src/main/scala/chisel3/tester/ChiselUtestTester.scala:
```diff
@@ -0,0 +1,74 @@
+  val topFileName = Some(getClass.getSimpleName)
+
+  /**
+   * Since [[utest.test]] is collided with [[chisel3.tester.RawTester.test]], it is renamed to [[testCircuit]],
```
```suggestion
   * Since [[utest.test]] collides with [[chisel3.tester.RawTester.test]], it has been aliased to [[testCircuit]].
```

edwardcwang commented on src/main/scala/chisel3/tester/ChiselUtestTester.scala:
```diff
@@ -0,0 +1,74 @@
+   * @note This API is experimental and forward compatibility is not yet guaranteed
+   * @param dutGen A generator of a Chisel Module
+   * @tparam T The DUT type, must be a subclass of MultiIOModule
+   * @return
```
```suggestion
```

edwardcwang commented on src/main/scala/chisel3/tester/ChiselUtestTester.scala:
```diff
@@ -0,0 +1,74 @@
+/**
+ * Using utest as test framework
+ * {{{
+ *   // define test spec in trait
```
I would suggest putting this code into a unit test

sequencer replied:
It's a little tricky, since we wanna use scalatest to test utest framework.

sequencer replied:
Seems not too tricky, here is the result:
```
[info] ScalaTest
[info] Run completed in 40 seconds, 304 milliseconds.
[info] Total number of tests run: 88
[info] Suites: completed 26, aborted 0
[info] Tests: succeeded 88, failed 0, canceled 0, ignored 1, pending 0
[info] All tests passed.
[info] utest
[info] Tests: 1, Passed: 1, Failed: 0
```

edwardcwang commented on build.sbt:
```diff
@@ -41,7 +41,8 @@ resolvers ++= Seq(
 libraryDependencies ++= Seq(
-  "org.scalatest" %% "scalatest" % "3.0.5"
+  "org.scalatest" %% "scalatest" % "3.0.5",
+  "com.lihaoyi" %% "utest" % "0.7.1"
```
@chick / @ucbjrl This should be OK?

sequencer commented on build.sbt:
```diff
@@ -41,8 +41,12 @@ resolvers ++= Seq(
+  "com.lihaoyi" %% "utest" % "0.7.1"
 )
+
+testFrameworks += new TestFramework("utest.runner.Framework")
```
Additional line to let `sbt test` invoke `utest`.

sequencer commented:
Yes, `RawTester` do work, the original intention of this PR is trying to fix the `test` function colliding of `RawTester.test` and `utest.test`, extending `ChiselUtestTester` will provide a better interface for `utest` users, avoiding import renaming.

ucbjrl commented:
Unfortunately, there doesn't appear to be a Scala 2.11 version of utest 07.1, which is causing current CI builds to fail:
```
[info] Setting Scala version to 2.11.12 on 1 projects.
[info] Reapplying settings...
[info] Set current project to chisel-testers2 (in build file:/vm/home/jenkins/sharedspace/chisel3-full/testers2/)
[info] Wrote /vm/home/jenkins/sharedspace/chisel3-full/testers2/target/scala-2.11/chisel-testers2_2.11-0.1-SNAPSHOT.pom
[warn] 
[warn] 	Note: Unresolved dependencies path:
[error] sbt.librarymanagement.ResolveException: Error downloading com.lihaoyi:utest_2.11:0.7.1
[error]   Not found
[error]   Not found
[error]   not found: /vm/home/jenkins/sharedspace/chisel3-full/ivy2/local/com.lihaoyi/utest_2.11/0.7.1/ivys/ivy.xml
[error]   not found: https://repo1.maven.org/maven2/com/lihaoyi/utest_2.11/0.7.1/utest_2.11-0.7.1.pom
[error]   not found: https://oss.sonatype.org/content/repositories/snapshots/com/lihaoyi/utest_2.11/0.7.1/utest_2.11-0.7.1.pom
[error]   not found: https://oss.sonatype.org/content/repositories/releases/com/lihaoyi/utest_2.11/0.7.1/utest_2.11-0.7.1.pom
```
I'll file an issue on https://github.com/lihaoyi/utest

sequencer commented:
Oh sorry. I didn’t remember we have to support 2.11, I’ll try to pr that later.

sequencer commented:
I just found that they
```
Dropped support for Scala 2.10.x, 2.11.x
```
in `0.6.9`
Emmm, maybe we can unlock utest to `0.6.8` until we dropped `2.11.x`?
